### PR TITLE
Refactor solver and move to internal

### LINKED ISSUE #73: Enable Deppy to be imported by other packages
Move all interfaces and broader implementations of those interfaces from `internal` to `pkg` 
Only keep internal implementation details of the interfaces in `internal`

**Make Deppy importable**
- [ ] Keep [solver](https://github.com/operator-framework/deppy/tree/main/internal/solver) package in `internal`
- [ ] Export [sat/constraints.go](https://github.com/operator-framework/deppy/blob/main/internal/sat/constraints.go) and [sat/variable.go](https://github.com/operator-framework/deppy/blob/main/internal/sat/variable.go)

### PR DESCRIPTION
**Pull request by perdasilva**
This PR refactors the sat solver and makes it possible to hide it in the internal package

Closes #73

### CODE REVIEW

**Comment by tmshort** [comment]
`sanity/lint` is failing

**Review by awgreene — COMMENTED** [review]
/lgtm

File: internal/solver/solve_test.go [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+			Variables: []deppy.Variable{
+				variable("a", constraint.Mandatory(), constraint.Dependency("y")),
+				variable("b", constraint.Mandatory(), constraint.Dependency("x", "y")),
+				variable("x"),
+				variable("y"),
+			},
+			Installed: []deppy.Identifier{"a", "b", "y"},
+		},
+		{
+			Name: "foo two dependencies satisfied by one variable",
```
Comment (oceanc80) [inline internal/solver/solve_test.go:255]: Maybe remove the test above in favor of this one?

Comment (perdasilva) [reply internal/solver/solve_test.go:245]: nice! fixed ^^

**Pull request merged**